proto/regen/ecocredit/v1/state.proto

```diff
@@ -381,3 +387,55 @@ message AllowedBridgeChain {
   // chain_name is the name of the chain allowed to bridge ecocredits to.
   string chain_name = 1;
 }
+
+// ClassApplication stores the data for a credit class application.
+message ClassApplication {
+  option (cosmos.orm.v1.table) = {
+    id : 17
+    primary_key : {fields : "id", auto_increment : true}
+    index : {id : 1, fields : "project_key", unique : true}
+  };
+
+  // id is the unique identifier of the application.
+  uint64 id = 1;
+
+  // project_key is the table row identifier of the project used internally for
+  // efficient lookups. This links an application to a project.
+  uint64 project_key = 2;
+
+  // class_key is the table row identifier of the credit class used internally
+  // for efficient lookups. This links a project to a credit class.
+  uint64 class_key = 3;
+
+  // issuer is the issuer of the credit class which the application has been
+  // submitted to.
+  string issuer = 4;
+
+  // metadata is any arbitrary metadata attached to the application.
+  string metadata = 5;
+
+  // status is the status of the application. Note that accepted and rejected
+  // applications are removed from the table.
+  ApplicationStatus status = 6;
+}
+
+// Application represents the evaluation status that a credit class issuer
+// assigns to a credit class application.
+enum ApplicationStatus {
```

Comment:
So possibly we should add an `APPLICATION_STATUS_ASSIGNED` if we are treating applications as a first come first serve by the issuers?

Though in seeing this, I'm wondering if it overcomplicates the process and if we would be better off just choosing either:
- projects apply to a specific issuer
- project applies to the credit class, and any issuer can change the status at any point in time (e.g. issuer A can set it as 'change requested' and issuer B can later accept the project)

Reply:
Currently I'm leaning towards the latter for simplicity